test/useLocalStorage.test.tsx

```diff
@@ -197,4 +234,20 @@ describe("useLocalStorage", () => {
     fireEvent(window, createStorageEventOption("username", null));
     expect(container.querySelector("p")).toHaveTextContent("");
   });
+  it("should remove item from localStorage when value is set as undefined", () => {
+    const { container } = render(<TestComponent />);
+
+    fireEvent.click(container.querySelector("#remove-data")!);
+    expect(container.querySelector("p")).toHaveTextContent("");
+    expect(localStorage.getItem("username")).toBe(null);
```

Comment:
Should the check be for `undefined` instead?
```suggestion
    expect(localStorage.getItem("username")).toBe(undefined);
```

Reply:
We are checking here if no value resides in localStorage for the key "username" so we should expect it to be `null`

Reply:
> Should the check be for `undefined` instead?

The test here is supposed to check if calling `setData(undefined)` will  as a side-effect cause the key `username` to be removed, in which i case i think it is entirely reasonable to expect it be `null` henceforth.